tsconfig.json

```diff
@@ -5,12 +5,12 @@
     "forceConsistentCasingInFileNames": true,
     "lib": ["ES2020"],
     "module": "CommonJS",
-    "moduleResolution": "node",
+    "moduleResolution": "nodenext",
```

Comment:
Good question. I also realize I didn't provide any context around this change at all.

There are functions in `@metamask/utils` which can only be imported in a Node.js context. The way you import these function is by importing `@metamask/utils/node`, not `@metamask/utils`. This path is defined as an export here: https://github.com/MetaMask/utils/blob/fb9440203099538670a4af95da16265f878f2cc9/package.json#L21

However, TypeScript does not know how to resolve this path because using a `moduleResolution` of `node` places TypeScript's module resolution logic in "classic" mode, where it only looks at the `types` field in `package.json` when resolving an import. Therefore it completely skips the `exports` field.

By setting `moduleResolution` to `nodenext`, we switch to the "modern" resolution logic, which does look at the `exports` field.

The `module` option is different, because that governs how TypeScript transforms imports and exports. Right now we have to use the CommonJS format because we know it works with the extension and mobile app. We cannot write any code that relies on ESM because neither the extension nor mobile app support this. From what I understand from reading the [docs around this](https://www.typescriptlang.org/docs/handbook/modules/reference.html#the-module-compiler-option), using a `module` of `CommonJS` guarantees that all of the code that we write will work under CommonJS, and this applies to dependencies as well.

Reply:
Changed here: [f4b8f8b](https://github.com/MetaMask/module-lint/pull/16/commits/f4b8f8bc6b84f26db60e6a04c82b7e2339e1f5cd)